dashboard/src/pages/TreeDetails/Tabs/Tests/IndividualTestsTable.tsx

```diff
@@ -151,57 +191,66 @@ export function IndividualTestsTable({
   const tableHeaders = useMemo((): JSX.Element[] => {
     return groupHeaders.map(header => {
       return (
-        <TableHead
+        <TH
           key={header.id}
-          className="border-b px-2 font-bold text-black"
+          className="px-2"
+          style={{ width: header.getSize() }}
         >
           {header.isPlaceholder
             ? null
             : flexRender(header.column.columnDef.header, header.getContext())}
-        </TableHead>
+        </TH>
       );
     });
     // TODO: remove exhaustive-deps and change memo  (all tables)
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [groupHeaders, sorting]);
 
-  const modelRows = table.getRowModel().rows;
+  const { rows } = useMemo(() => {
+    return table.getRowModel();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [table, sorting]);
+
+  const parentRef = useRef<HTMLDivElement>(null);
+  const virtualizer = useVirtualizer({
+    count: rows.length,
+    estimateSize: () => ESTIMATED_ROW_HEIGHT,
+    getScrollElement: () => parentRef.current,
+    overscan: 5,
+  });
+  const virtualItems = virtualizer.getVirtualItems();
+
   const tableRows = useMemo((): JSX.Element[] => {
-    return modelRows?.length
-      ? modelRows.map(row => (
-          <TableRow
-            key={row.id}
-            data-state={row.getIsSelected() && 'selected'}
-            className="cursor-pointer hover:bg-lightBlue"
-          >
-            {row.getVisibleCells().map(cell => (
-              <TableCellWithLink
-                key={cell.id}
-                linkProps={{
-                  to: '/tree/$treeId/test/$testId',
-                  params: {
-                    testId: row.original.id,
-                  },
-                  search: s => s,
-                }}
-              >
-                {flexRender(cell.column.columnDef.cell, cell.getContext())}
-              </TableCellWithLink>
-            ))}
-          </TableRow>
-        ))
-      : [
-          <TableRow key="no-results">
-            <TableCell colSpan={columns.length} className="h-24 text-center">
-              <FormattedMessage id="global.noResults" />
-            </TableCell>
-          </TableRow>,
-        ];
-  }, [modelRows]);
+    return virtualItems.map(virtualRow => {
+      const row = rows[virtualRow.index] as Row<TIndividualTest>;
+      return <TableRowMemoized row={row} key={row.id} />;
+    });
+  }, [rows, virtualItems]);
+
+  const [paddingTop, paddingBottom] = useMemo(() => {
+    if (virtualItems.length === 0) return [0, 0];
+    return [
+      virtualItems[0].start,
+      virtualizer.getTotalSize() - virtualItems[virtualItems.length - 1].end,
+    ];
```

Comment:
if you return the objects with style from this useMemo you will get a stable reference